src/beliefpropagation/beliefpropagation_schedule.jl

```diff
@@ -0,0 +1,40 @@
+default_edge_sequence_alg() = "forest_cover"
+
+@traitfn default_bp_niters(g::NamedGraph::(!IsDirected)) = is_tree(g) ? 1 : nothing
```

Comment:
```suggestion
@traitfn default_bp_niters(g::::(!IsDirected)) = is_tree(g) ? 1 : nothing
```
Can just be for any undirected graph.

Reply:
This seems to result in an infinite loop and the trait function never gets called?
```
@traitfn default_bp_niters(g::NamedGraph::(!IsDirected)) = is_tree(g) ? 1 : nothing
function default_bp_niters(g::AbstractGraph)
  return default_bp_niters(undirected_graph(underlying_graph(g)))
end
```

Reply:
Also should we really allow the trait function to dispatch on any type which is not directed or should we at least constraint it to abstractgraphs via
`@traitfn default_bp_niters(g::AbstractGraph::(!IsDirected)) = is_tree(g) ? 1 : nothing`

Reply:
What about

```
@traitfn undirected_default_bp_niters(g::AbstractGraph::(!IsDirected)) =
  is_tree(g) ? 1 : nothing
function default_bp_niters(g::AbstractGraph)
  return undirected_default_bp_niters(undirected_graph(underlying_graph(g)))
end
```